geometric/optimize.py

```diff
@@ -502,36 +711,8 @@ def step(self):
             ##### End Rebuild
             # Finally, take an internal coordinate step of the desired length.
             dy, _ = self.trust_step(iopt, v0, verbose=(self.params.verbose+1 if self.params.verbose >= 2 else 0))
-            self.cnorm = self.get_cartesian_norm(dy)
-        ### DONE OBTAINING THE STEP ###
-        if isinstance(self.IC, PrimitiveInternalCoordinates):
-            idx = np.argmax(np.abs(dy))
-            iunit = np.zeros_like(dy)
-            iunit[idx] = 1.0
-            self.prim_msg = "Along %s %.3f" % (self.IC.Internals[idx], np.dot(dy/np.linalg.norm(dy), iunit))
-        ### These quantities, computed previously, are no longer used.
-        # Dot product of the gradient with the step direction
-        # Dot = -np.dot(dy/np.linalg.norm(dy), self.G/np.linalg.norm(self.G))
-        # Whether the Cartesian norm comes close to the trust radius
-        # bump = cnorm > 0.8 * self.trust
-        ### Before updating any of our variables, copy current variables to "previous"
-        self.Yprev = self.Y.copy()
-        self.Xprev = self.X.copy()
-        self.Gxprev = self.gradx.copy()
-        self.Gprev = self.G.copy()
-        self.Eprev = self.E
-        ### Update the Internal Coordinates ###
-        X0 = self.X.copy()
-        self.newCartesian(dy)
-        ## The "actual" dy may be different from the one passed to newCartesian(),
-        ## for example if we enforce constraints or don't get the step we expect.
-        dy = self.IC.calcDiff(self.X, X0)
-        # dyp = self.IC.Prims.calcDiff(self.X, X0)
-        # print("Actual dy:", dy)
-        self.Y += dy
-        self.expect = flat(0.5*multi_dot([row(dy),self.H,col(dy)]))[0] + np.dot(dy,self.G)
-        # self.expectdG = np.dot(self.H, col(dy).flatten())
-        self.state = OPT_STATE.NEEDS_EVALUATION
+
+        return dy
 
     def evaluateStep(self):
```

Comment:
Would it be possible to have separate functions for `evaluate_optimize_step()` and `evaluate_irc_step()` keeping the "common" codes in this function, similar to how `step()` was organized?

Reply:
I attempted to separate them. You can see the commit [here](https://github.com/leeping/geomeTRIC/pull/197/commits/2df277123c6ead67ebc706aca67aebca432796bd).